orangecontrib/text/widgets/ownyt.py

```diff
@@ -10,14 +10,14 @@
 from orangecontrib.text.corpus import Corpus
 from orangecontrib.text.nyt import NYT, MIN_DATE
 from orangecontrib.text.widgets.utils import CheckListLayout, DatePickerInterval, QueryBox, \
-    gui_require, OWConcurrentWidget, asynchronous
+    gui_require, asynchronous
 
 
 class IO:
     CORPUS = "Corpus"
 
 
-class OWNYT(OWConcurrentWidget):
+class OWNYT(OWWidget):
```

Comment:
The third option would be to remove `should_break` calls and let the `callback` raise `StopExecution`. Then all the called method could also have theirs try except clauses to handle StopExecution as preferred.